accepted/better-obsoletion/better-obsoletion.md

```diff
@@ -0,0 +1,345 @@
+often than not the API that looks the most promising might be the wrong one to
+use. So it becomes important that we can point developers the right way.
+
+Since .NET Framework 1.0, we were able to mark APIs as obsoleted via a built-in
+attribute. Any usage of such an API results in the compiler emitting a
+diagnostic telling the developer that the API is obsoleted, which can include a
+small piece of text pointing to an alternative.
+
+This obsoletion approach has a few benefits over the alternatives:
+
+* **API Removal**. Removing APIs is a very disrupting event. Some argue that
+  SemVer solves this problem but it doesn't work well for platforms as it ties
+  unrelated decisions together (e.g. in order to get access to new APIs, an
+  unrelated area in the code needs to be rewritten to avoid using the removed
+  APIs). Other technologies, such as C++, can use the preprocessor to remove
+  features unless the customer explicitly opts-in to legacy behavior. This
+  doesn't work well for .NET as this ecosystem is built around the reuse of
+  binaries, not source code.
+
+* **Documentation**. Let's face it, developers only read documentation when
+  something doesn't work. Only few developers research guidance and
+  best-practice proactively. But even if they did: guidance changes over time
+  and some technologies that looked like good ideas turn out to be unviable. We
+  need a mechanism that informs developers after the fact.
+
+Obsoletion via the `[Obsolete]` attribute works pretty well when: only a few
+APIs are affected, it's easy to remove their usage, and guidance can be provided
+as a short one-line string.
+
+In practice these requirements are often not satisfiable:
+
+* **Technologies have large API surfaces**. Obsoleting an entire technology
+  means that we'll have to annotate all entry points to that technology, which
+  often means obsoleting a *set of APIs*, such as types and static members.
+
+* **Removing obsoleted APIs isn't always practical**. It would be nice if we
+  could always remove dependencies on obsoleted technologies but this isn't
+  always possible if they are used heavily in existing code bases. In those
+  cases, you typically want a way to ignore obsoletion diagnostics regarding
+  specific technologies but not others so that you don't accidentally start
+  depending on obsoleted technologies when you're evolving your code base.
+
+* **Guidance is complex**. Obsoleted APIs rarely have a drop-in replacement.
+  After all, we don't obsolete based on their naming or shape but usually due to
+  fundamental design issues. That means the replacements are typically a nuanced
+  choice and require at least a page of explanation, rather than a one-liner. 
+
+This document proposes changes to how APIs can be obsoleted in a more structured
+way.
+
+## Scenarios and User Experience
+
+### Targeted Suppression
+
+Abigail is a developer and maintains the business logic layer component for
```

Comment:
Presumably this is a fictitious example, correct? Add a preamble to clarify that.

Reply:
All specs follow that pattern. The fact that you’ve never seen this makes me sad because it means we no longer write proper specs 😬

Reply:
Or, I haven't read enough of them :)

Good to know.